pkg/parser/boolevator/boolevator.go

```diff
@@ -44,6 +45,11 @@ func (boolevator *Boolevator) Eval(expr string, env map[string]string) (bool, er
 	// Work around text/scanner stopping at newline when scanning strings
 	expr = strings.ReplaceAll(expr, "\n", "\\n")
 
+	// Pre-expand environment variables
+	for key, value := range env {
+		env[key] = expander.ExpandEnvironmentVariables(value, env)
```

Comment:
Let's create a new map so the `Eval` won't have side effects.

Reply:
Good catch! Fixed in e519dd374153223383894d50b8415f2b287dabea.